packages/file_selector/file_selector_platform_interface/lib/src/method_channel/method_channel_file_selector.dart

```diff
@@ -0,0 +1,65 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:typed_data';
+
+import 'package:flutter/services.dart';
+
+import '../platform_interface/file_selector_interface.dart';
+import 'package:file_selector_platform_interface/file_selector_platform_interface.dart';
+
+const MethodChannel _channel = MethodChannel('plugins.flutter.io/file_picker');
+
+/// An implementation of [FileSelectorPlatform] that uses method channels.
+class MethodChannelFileSelector extends FileSelectorPlatform {
+  /// Load a file from user's computer and return it as an XFile
+  @override
+  Future<XFile> loadFile({
+    List<XTypeGroup> acceptedTypeGroups,
+    String initialDirectory,
+    String confirmButtonText,
+  }) {
+    return _channel.invokeMethod<XFile>(
```

Comment:
This will probably not work, because the method channels cannot transport "complex" responses without specifying a custom Binary messenger (which... I don't want to do :P).

Most likely this will have to do:

```dart
final path = await _channel.invokeMethod<String>('loadFile'...)
return XFile(path: path);
```